base/primes.jl

```diff
@@ -11,18 +11,28 @@ function primesmask(s::AbstractVector{Bool})
     n < 2 && return s; s[2] = true
     n < 3 && return s; s[3] = true
     r = isqrt(n)
-    for x = 1:r
-        xx = x*x
-        for y = 1:r
-            yy = y*y
-            i, j, k = 4xx+yy, 3xx+yy, 3xx-yy
-            i <= n && (s[i] $= (i%12==1)|(i%12==5))
-            j <= n && (s[j] $= (j%12==7))
-            1 <= k <= n && (s[k] $= (x>y)&(k%12==11))
+    for x = 1:r; x² = x*x
+        for y = 1:r; y² = y*y
+            i = 4x² + y²
+            if i ≤ n && (i % 12 == 1 || i % 12 == 5)
+                @inbounds s[i] = !s[i]
+            end
+            j = 3x² + y²
+            if j ≤ n && j % 12 == 7
+                @inbounds s[j] = !s[j]
+            end
+            k = 3x² - y²
+            if x > y && k ≤ n && k % 12 == 11
+                @inbounds s[k] = !s[k]
+            end
         end
     end
-    for i = 5:r
-        s[i] && (s[i*i:i*i:n] = false)
+    for i = 5:r; i² = i*i
+        @inbounds if s[i]
+            for j = i²:i²:n
+                s[j] = false
+            end
+        end
```

Comment:
I don't think there's need to change this as it won't make a difference and the current form is really compact. You might keep the `@inbounds` though (at the beginning of the for loop).